### PR TITLE
ARROW-1410: Remove MAP_POPULATE flag when mmapping files in Plasma store.

### PR DESCRIPTION
**Pull request by robertnishihara**
cc @pcmoritz @atumanov

### CODE REVIEW

**Comment by pcmoritz** [comment]
The reasoning for removing this is the following one: MAP_POPULATE will pre-populate the page entries when the mmap is executed. While this speeds up accessing the file from the client, it also induces a large latency penalty when a new memory mapped file is created. This latency can be on the order of several 10s of seconds for very large memory mapped files and we would like to avoid this behavior which can make the object store performance unpredictable.

File: cpp/src/plasma/malloc.cc [hunk 1]
```diff
@@ -135,10 +135,10 @@ void* fake_mmap(size_t size) {
   int fd = create_buffer(size);
   ARROW_CHECK(fd >= 0) << "Failed to create buffer during mmap";
 #ifdef __linux__
```
Comment (pcmoritz) [inline cpp/src/plasma/malloc.cc]: Let's get rid of this ifdef now; we can still have the comment that explains why we don't use mmap.

Comment (robertnishihara) [reply cpp/src/plasma/malloc.cc]: good idea, done

**Comment by atumanov** [comment]
the difference in performance behavior with and without MAP_POPULATE is the following: with MAP_POPULATE perf degradation occurs (and is tangible) *only when* we run out of space to place a new object and, thus, need to mmap a new file. Without MAP_POPULATE, we distribute *degraded* performance (due to page faulting on the server side) uniformly across most object puts. As stated, the advantage of the latter is throughput predictability.

It is important to note that, in steady state, when the object store has been fully warmed up, presence/absence of MAP_POPULATE doesn't matter. It only matters during the object store ramp up.

My favorite longer-term solution continues to be preallocating the configured capacity on object store startup as a single mmapped file and pre-faulting it in the background.

**Comment by pcmoritz** [comment]
+1, the build failure in AppVeyor is unrelated

**Pull request closed without merging**